### PR TITLE
Refresh Token, Web/Public App support

### PR DESCRIPTION
**Pull request by sfcbetiuc**
* Added Refresh Token support for OAuth2 authentication
* Added Web/Public App support for OAuth2 authentication

### CODE REVIEW

File: FuelSDK/client.py [hunk 1]
```diff
@@ -50,8 +53,13 @@ def __init__(self, get_server_wsdl = False, debug = False, params = None, tokenR
         else:
             logging.getLogger('suds').setLevel(logging.INFO)
 
+        self.set_class_variables(get_server_wsdl, params, tokenResponse)
+
+    def set_class_variables(self, get_server_wsdl, params, tokenResponse):
```
Comment (sfdrogojan) [inline FuelSDK/client.py]: Can this be renamed to *configure_client*?

File: README.md [hunk 2]
```diff
@@ -1,11 +1,64 @@
...
+        'useOAuth2Authentication': 'True',
+        'accountId': '<TARGET_ACCOUNT_ID>',
+        'scope': '<PERMISSION_LIST>'
+        'applicationType': '<APPLICATION_TYPE>'
+        'redirectURI': '<REDIRECT_URI_FOR_PUBLIC/WEB_APP>'
+        'authorizationCode': '<AUTHORIZATION_CODE_FOR_PUBLIC/WEB_APP>'
+    })
+  ```
+  
+* applicationType will have one of the following values: `server`, `public`, `web`. The default value of applicationType is `server`.
```
Comment (sfdrogojan) [inline README.md]: Replace 'will' with 'can'.

File: FuelSDK/client.py (hunk 1 above)
Comment (sfcbetiuc) [reply FuelSDK/client.py]: Done

File: README.md (hunk 2 above)
Comment (sfcbetiuc) [reply README.md]: Done

File: .gitignore [hunk 3]
```diff
@@ -31,6 +31,8 @@ objsamples/ET_Client.py
 objsamples/config.python
 objsamples/*.pyc
 config.python
+FuelSDK/Public_WebAppTests/config.python
```
Comment (sfdrogojan) [inline .gitignore]: Please add template file for the config needed to run the tests in this folder.

Comment (sfcbetiuc) [reply .gitignore]: Template added.

**Pull request merged**